mimic/test/behavior_tests.py

```diff
@@ -0,0 +1,213 @@
+"""
+Automatically generate tests for behavior registration/deletion APIs.
+Also contains helper functions for specific behavior testing.
+"""
+import json
+
+from uuid import UUID, uuid4
+
+from six import text_type, string_types
+
+from twisted.trial.unittest import SynchronousTestCase
+
+from zope.interface import Attribute, Interface
+
+from mimic.test.helpers import json_request, request_with_content
+
+
+class IBehaviorAPITestHelper(Interface):
+    """
+    Helper class that provides some setup and assertion methods that tests
+    for behavior CRUD need.  An instance will be created and
+    used for each CRUD test method.
+    """
+    root = Attribute("The root resource for mimic.")
+    behavior_api_endpoint = Attribute(
+        "The API endpoint for behaviors for the event to be tested.")
+    criteria = Attribute("""
+    A list of criteria for the behaviors to be injected.  Example::
```

Comment:
This is a silly nitpick (let's not address it now) but the docstrings for `Attribute` should probably be indented so they're not flush with the indentation level where they're being declared.